spark/src/main/scala/geotrellis/spark/io/s3/S3AttributeStore.scala

```diff
@@ -33,6 +33,9 @@ class S3AttributeStore(bucket: String, rootPath: String) extends AttributeStore[
 
   def path(parts: String*) = parts.filter(_.nonEmpty).mkString("/")
 
+  def attributePath(id: LayerId): String =
+    path(rootPath, "_attributes", s"*__${id.name}__${id.zoom}.json")
```

Comment:
I would expect `readAll` to be used more often than delete. That means the only thing you can do is list all the keys in the attribute store, filter for this layer and delete those.